PolyEngine/UnitTests/Src/SafePtrTests.cpp

```diff
@@ -0,0 +1,20 @@
+#include <catch.hpp>
+
+#include <Defines.hpp>
+#include <SafePtrRoot.hpp>
+#include <SafePtr.hpp>
+
+using namespace Poly;
+
+TEST_CASE("Safe pointer tests", "[SafePtr]") {
+	SafePtrRoot safePtrObject = SafePtrRoot();
+	SafePtr<SafePtrRoot> safePtr = SafePtr<SafePtrRoot>(&safePtrObject);
+
+	REQUIRE(safePtr->Debug() == safePtrObject.Debug());
```

Comment:
You can provide this debug method by implementing new test class that inherits from SafePtrRoot.